compiler/src/dotty/tools/dotc/transform/SyntheticMethods.scala

```diff
@@ -261,12 +261,30 @@ class SyntheticMethods(thisPhase: DenotTransformer) {
      */
     def canEqualBody(that: Tree): Tree = that.isInstance(AnnotatedType(clazzType, Annotation(defn.UncheckedAnnot)))
 
-    symbolsToSynthesize flatMap syntheticDefIfMissing
+    val methods = symbolsToSynthesize flatMap syntheticDefIfMissing
+
+    def createReadResolveMethod: Tree = {
```

Comment:
It'd be nice to extract the common code between this method and `syntheticDef` in a separate method somehow.